packages/aragon-wrapper/src/index.js

```diff
@@ -596,14 +608,14 @@ export default class Aragon {
     // Get the application proxy
     // Only get the first result from the observable, so our running contexts don't get
     // reinitialized if new apps appear
-    const appProxy = await this.apps
-      .map((apps) => {
+    const appProxy = await this.apps.pipe(
+      map((apps) => {
         const app = apps.find((app) => addressesEqual(app.proxyAddress, proxyAddress))
         // TODO: handle undefined (no proxy found), otherwise when calling app.proxyAddress next, it will throw
         return makeProxyFromABI(app.proxyAddress, app.abi, this.web3)
-      })
-      .take(1)
-      .toPromise()
+      }),
+      take(1)
+    ).toPromise()
```

Comment:
Maybe we could take this entire bit out of the observables.

What we currently do:

1. From `apps` observable, map each emission
1. When mapping each emission, get an `AppProxy` out
1. Oh wait—we only want the first emission

What do you think about changing it to:

1. Get first emission of `apps` observable (since it's `publishReplay()`'d, it's the current value) and turn it into a promise that resolves with an array
1. Do a normal operations with the array to `find()` and get an `AppProxy`

This way it'd be much easier to emit errors from here in the future as well.